torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4471,19 +4485,27 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            supports_inplace_autograd=False,
            sample_inputs_func=sample_inputs_kron),
     OpInfo('inner',
-           dtypes=floating_types(),
-           dtypesIfCPU=all_types_and_complex_and(torch.float16, torch.bfloat16),
-           # BFloat16 support on CUDA requires CUDA 11 and SM53
-           dtypesIfCUDA=floating_types_and(torch.float16, torch.complex64, torch.complex128,
-                                           *[torch.bfloat16] if CUDA11OrLater else []),
-           dtypesIfROCM=floating_types_and(torch.half),
-           supports_out=True,
-           sample_inputs_func=sample_inputs_inner,
+           dtypes=floating_and_complex_types_and(torch.half),
+           dtypesIfCPU=all_types_and_complex_and(torch.half, torch.bfloat16),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.float16, *[torch.bfloat16] if CUDA11OrLater else []),
+           dtypesIfROCM=floating_and_complex_types_and(torch.half, torch.bfloat16),
+           sample_inputs_func=sample_inputs_inner),
+    OpInfo('tensordot',
+           dtypes=floating_and_complex_types_and(torch.half),
+           dtypesIfCPU=all_types_and_complex_and(torch.half, torch.bfloat16),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.float16, *[torch.bfloat16] if CUDA11OrLater else []),
+           dtypesIfROCM=floating_and_complex_types_and(torch.half, torch.bfloat16),
+           safe_casts_outputs=True,
+           sample_inputs_func=sample_inputs_tensordot,
            skips=(
-               # Reference Issue: https://github.com/pytorch/pytorch/issues/56022
-               # AssertionError: UserWarning not triggered : Resized a non-empty tensor but did not warn about it.
-               SkipInfo('TestCommon', 'test_out', dtypes=[torch.float32]),
-           )),
+               # Currently failing due to an INTERNAL_ASSERT_FAILED error.
+               # Reference: https://github.com/pytorch/pytorch/issues/56314
+               SkipInfo("TestCommon", "test_variant_consistency_jit", dtypes=[torch.float32]),
```

Comment:
Maybe `tensordot` doesn't support JIT.

Reply:
Since `test_variant_consistency_jit` passes for `complex64`, could it be the case that `tensordot` is actually jittable?
If so, perhaps `check_alias_annotation` shouldn't be called for functionals such as `tensordot` in `test_variant_consistency_jit`:
https://github.com/pytorch/pytorch/blob/5cab3b9cf628ba44b599fa7da4c06404986dd656/test/test_ops.py#L397-L403

Reply:
`tensordot` has been jittable since #53672, so I won't change the comment with the `SkipInfo`.